### DISCUSSION TITLE
How to achieve avalonia browser program and server communication

### DISCUSSION THREAD

**Discussion by zy1075984** [AvaloniaUI/Avalonia#11921] [Q&A] [answered]
How to achieve avalonia browser program and server communication through WebSocket, can you provide a complete example, thank very much！

**Comment by kyuranger** [comment, marked as answer]
Maybe you can use the SignalR:
https://learn.microsoft.com/en-us/aspnet/core/signalr/introduction?view=aspnetcore-7.0

**Comment by maxkatz6** [comment]
Use build-in .NET APIs for HttpClient or WebSockets. It has nothing to do with Avalonia.
But the browser itself does limit your choices, as there is no support for raw sockets directly.
